telemetry-core/src/main/java/com/newrelic/telemetry/transport/BatchDataSender.java

```diff
@@ -124,7 +141,11 @@ private Response sendPayload(byte[] payload, UUID requestId, String batchType)
       throws DiscardBatchException, RetryWithSplitException, RetryWithBackoffException,
           RetryWithRequestedWaitException {
     Map<String, String> headers = new HashMap<>();
-    headers.put("Api-Key", apiKey);
+    if (useLicenseKey) {
+      headers.put("X-License-Key", apiKey);
+    } else {
+      headers.put("Api-Key", apiKey);
+    }
```

Comment:
You could avoid having to do this if / else every time you send the payload by initializing some field with the name of the api key header at construction time. I.e.:
```
this.apiKeyHeader = useLicenseKey ? "X-License-Key" : "Api-Key";

....
Map<String, String> headers = new HashMap<>();
headers.put(this.apiKeyHeader, apiKey);
```